indico/web/client/js/react/components/principals/items.jsx

```diff
@@ -39,6 +35,92 @@ PendingPrincipalListItem.defaultProps = {
   type: PrincipalType.user,
 };
 
+const PrincipalItemIcon = ({type, meta, invalid, avatarURL, className}) =>
+  type === PrincipalType.eventRole || type === PrincipalType.categoryRole ? (
+    <div styleName="event-role" className={className}>
+      <span style={meta.style}>{meta.code}</span>
+    </div>
+  ) : (
+    <div styleName="icon">
+      {invalid ? (
+        <Popup
+          trigger={
+            <Icon.Group size="large">
+              <Icon name={PrincipalType.getIcon(type)} />
+              <Icon name="exclamation triangle" color="orange" corner />
+            </Icon.Group>
+          }
+        >
+          {PrincipalType.getDeletedText(type)}
+        </Popup>
+      ) : avatarURL ? (
+        <Image src={avatarURL} className={className} size="mini" avatar />
+      ) : (
+        <Icon name={PrincipalType.getIcon(type)} size="large" />
+      )}
+    </div>
+  );
+
+PrincipalItemIcon.propTypes = {
+  type: PropTypes.string,
+  meta: PropTypes.object,
+  invalid: PropTypes.bool,
+  avatarURL: PropTypes.string,
+  className: PropTypes.string,
+};
+
+PrincipalItemIcon.defaultProps = {
+  type: PrincipalType.user,
+  meta: {},
+  invalid: false,
+  avatarURL: null,
+  className: undefined,
+};
+
+const PrincipalItemContent = ({name, detail, children}) => (
+  <div styleName="content">
+    <List.Content>{name}</List.Content>
+    {detail && (
+      <List.Description>
+        <small>{detail}</small>
+      </List.Description>
+    )}
+    {children}
+  </div>
+);
+
+PrincipalItemContent.propTypes = {
+  name: PropTypes.string.isRequired,
+  detail: PropTypes.string,
+  children: PropTypes.node,
+};
+
+PrincipalItemContent.defaultProps = {
+  detail: null,
+  children: null,
+};
+
+export const PrincipalItem = ({as, children, className, ...rest}) => {
+  const Component = as || 'div';
```

Comment:
Putting `as: 'div'` in `defaultProps` doesn't work?

Reply:
It does, even though it's a bit more loose.